server/src/main/java/org/apache/calcite/server/ServerDdlExecutor.java

```diff
@@ -558,6 +568,71 @@ public void execute(SqlCreateTable create,
     }
   }
 
+  /** Executes a {@code CREATE TABLE LIKE} command. */
+  public void execute(SqlCreateTableLike create,
+      CalcitePrepare.Context context) {
+    final Pair<CalciteSchema, String> pair = schema(context, true, create.name);
+    final JavaTypeFactory typeFactory = context.getTypeFactory();
+    final Table table = table(context, create.sourceTable);
+    final RelDataType rowType = table.getRowType(typeFactory);
+    InitializerExpressionFactory ief = NullInitializerExpressionFactory.INSTANCE;
+    if (table instanceof Wrapper) {
+      final InitializerExpressionFactory sourceIef =
+          ((Wrapper) table).unwrap(InitializerExpressionFactory.class);
+      if (sourceIef != null) {
+        final Set<SqlCreateTableLike.LikeOption> optionSet = create.options();
+        final boolean includingGenerated =
+            optionSet.contains(SqlCreateTableLike.LikeOption.GENERATED)
+                || optionSet.contains(SqlCreateTableLike.LikeOption.ALL);
+        final boolean includingDefaults =
+            optionSet.contains(SqlCreateTableLike.LikeOption.DEFAULTS)
+                || optionSet.contains(SqlCreateTableLike.LikeOption.ALL);
+        ief = new NullInitializerExpressionFactory() {
```

Comment:
Need class name and document, not anonymous class.

Reply:
Fixed.